### ISSUE TITLE
How to Generate Stored Procedure

### ISSUE THREAD

**Issue by roppongiosamu** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#565]
Hi.
I bought a commercial licence today.

I have a question.

I am using a version of .net core.
I succeed to generate source of POCO.
But I cant't generate Stored Procedure.

From the below documents,
```
Full control over the generated code
bility to control exactly what is generated. Such as POCO classes, DB Context, DB Context interface, POCO Configuration/Mapping classes, Stored procedures and Enums. This is required for enterprise software where different parts of the configuration can exist in different folders or projects

```
so, I  thought that I can generate Stored Procedures. 
Please teach me whether Stored Procedures can generate or not.

**Comment by sjh37** [comment]
Hi. Thanks for buying a license.
Yes stored procedures are fully reverse engineered and the EF calling code created for you to use.

The Settings.ConnectionString string you use must have at least these privileges: `ddladmin`, `datareader` and `datawriter`. `ddladmin` is required for reading the default constraints.

Stored procedure calls are placed into the database context. Have a search for `// Stored Procedures`

Also check your filtering, you may have filtered the stored procedures out.
Set the following to true:
```c#
FilterSettings.IncludeStoredProcedures      = true;
FilterSettings.IncludeTableValuedFunctions  = true;
FilterSettings.IncludeScalarValuedFunctions = true;
```

And save your `<dataabse>.tt` file and see if the stored procedures have been generated.

**Comment by roppongiosamu** [comment]
Thank you for your reply.
I could understand the feature of creating StoredProcedure. 
I was thinking that the feature create StoredProcedure automatically, but as a matter of fact, that means  you mentioned above.

> stored procedures are fully reverse engineered and the EF calling code created for you to use

I modified the source, then I succeded, but compile errors occured, i need the bellow action.

https://stackoverflow.com/questions/58347444/entity-framework-core-3-raw-sql-missing-methods
`install-package Microsoft.EntityFrameworkCore.SqlServer
`
